### PR TITLE
[7.x][ML] Explicitly require a OriginSettingClient in ML results iterators 

### PR DESCRIPTION
**Pull request by davidkyle**
In classes where the client is used directly rather than through a call to
executeAsyncWithOrigin explicitly require the client to be OriginSettingClient
rather than using the Client interface.
    
Also remove calls to deprecated ClientHelper.clientWithOrigin() method.

Backport of #50802

### CODE REVIEW

File: x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/job/retention/AbstractExpiredJobDataRemoverTests.java [hunk 1]
```diff
@@ -126,13 +140,14 @@ public void testRemoveGivenMultipleBatches() throws IOException {
 
         AtomicInteger searchCount = new AtomicInteger(0);
 
-        @SuppressWarnings("unchecked")
-        ActionFuture<SearchResponse> future = mock(ActionFuture.class);
-        doAnswer(invocationOnMock -> responses.get(searchCount.getAndIncrement())).when(future).actionGet();
-        when(client.search(any())).thenReturn(future);
+        doAnswer(invocationOnMock -> {
```
Comment (przemekwitek) [inline x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/job/retention/AbstractExpiredJobDataRemoverTests.java]: Could this snippet be replaced with an appropriate `mockSearchResponse` call?

File: x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/job/retention/ExpiredModelSnapshotsRemoverTests.java [hunk 2]
```diff
@@ -86,25 +91,22 @@ public void testRemove_GivenJobsWithoutRetentionPolicy() throws IOException {
 
         listener.waitToCompletion();
         assertThat(listener.success, is(true));
-        verify(client).search(any());
-        Mockito.verifyNoMoreInteractions(client);
```
Comment (przemekwitek) [inline x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/job/retention/ExpiredModelSnapshotsRemoverTests.java]: Why is this removed? Are there any new `client` interactions expected after your change?

File: x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/job/retention/ExpiredResultsRemoverTests.java [hunk 3]
```diff
@@ -43,58 +41,48 @@
...
-                 public Void answer(InvocationOnMock invocationOnMock) throws Throwable {
-                     capturedDeleteByQueryRequests.add((DeleteByQueryRequest) invocationOnMock.getArguments()[1]);
-                     ActionListener<BulkByScrollResponse> listener =
-                             (ActionListener<BulkByScrollResponse>) invocationOnMock.getArguments()[2];
-                     listener.onResponse(null);
-                     return null;
-                 }
-             }).when(client).execute(same(DeleteByQueryAction.INSTANCE), any(), any());
+
+        client = org.mockito.Mockito.mock(Client.class);
```
Comment (przemekwitek) [inline x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/job/retention/ExpiredResultsRemoverTests.java]: Could `org.mockito.Mockito.mock` be imported to avoid qualified name?

File: x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/test/MockOriginSettingClient.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.ml.test;
+
+
```
Comment (przemekwitek) [inline x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/test/MockOriginSettingClient.java]: Please remove blank line.

**Comment by davidkyle** [comment]
run elasticsearch-ci/1

**Review by przemekwitek — APPROVED** [review]
Looks good overall, just a few minor comments.

File: x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/job/retention/ExpiredModelSnapshotsRemoverTests.java (hunk 2 above)
Comment (davidkyle) [reply x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/job/retention/ExpiredModelSnapshotsRemoverTests.java]: Yes, the filter client makes many more calls to the inner client.
